src/opm/output/eclipse/Summary.cpp

```diff
@@ -3396,51 +3402,58 @@ namespace Evaluator {
 
     bool Factory::isFunctionRelation()
     {
-        auto pos = funs.find(this->node_->keyword);
+        const auto normKw = Opm::EclIO::SummaryNode::
+            normalise_keyword(this->node_->category, this->node_->keyword);
+
+        auto pos = funs.find(normKw);
         if (pos != funs.end()) {
             // 'node_' represents a functional relation.
             // Capture evaluation function and return true.
             this->paramFunction_ = pos->second;
             return true;
         }
 
-        auto keyword = this->node_->keyword;
-        auto dash_pos = keyword.find("_");
-        if (dash_pos != std::string::npos)
-            keyword = keyword.substr(0, dash_pos);
+        if (normKw.length() <= std::string::size_type{4}) {
+            return false;
+        }
+
+        const auto tracer_name = normKw.substr(4);
+
+        const auto& tracers = this->es_.tracer();
+        auto trPos = std::find_if(tracers.begin(), tracers.end(),
+                                  [&tracer_name](const auto& tracer)
+                                  {
+                                      return tracer.name == tracer_name;
+                                  });
+
```

Comment:
I think this is a change of semantics as we now assume that there is only one tracer per name. That seems to make sense, though.

On a tangent: The code in TracerConfig.cpp seems to add multiple entries if a tracer with the same name occurs twice. Not sure what the correct behaviour should actually (print an error, overwrite)? The rest of the code in TracerConfig also seems to assume that there cannot be multiple tracers with the same name.

Reply:
> I think this is a change of semantics as we now assume that there is only one tracer per name. That seems to make sense, though.

Tracer names are supposed to be unique, one name per tracer, and defined in the `TRACERS` keyword from which our `TracerConfig` class is constructed.  There is no change in semantics however because although the previous code might appear to select the **last** of multiple entries with the same name, we would in fact [pick an implementation function](https://github.com/OPM/opm-common/blob/c5158a1c24c3795ad67d759aadb3cfda228b4613/src/opm/output/eclipse/Summary.cpp#L3433-L3437) and return `true` on the first match.

> The code in TracerConfig.cpp seems to add multiple entries if a tracer with the same name occurs twice. Not sure what the correct behaviour should actually (print an error, overwrite)?

I'd be in favour of generating an error and stopping in this case, but we might want to discuss that with other developers.